modules/tinymce/src/core/test/ts/browser/html/DomParserTest.ts

```diff
@@ -151,71 +153,71 @@ describe('browser.tinymce.core.html.DomParserTest', () => {
     assert.deepEqual(countNodes(root), { 'body': 1, 'code': 1, '#text': 1 }, 'Whitespace inside code (count)');
   });
 
-  // TODO: TINY-4627/TINY-8202: Update tests to account for omitted </p>
-  it.skip('Parse invalid contents', () => {
+  it('Parse invalid contents', () => {
     let parser, root;
 
     parser = DomParser({}, schema);
     root = parser.parse('<p class="a"><p class="b">123</p></p>');
-    assert.equal(serializer.serialize(root), '<p class="b">123</p>', 'P in P, no nodes before/after');
-    assert.deepEqual(countNodes(root), { 'body': 1, 'p': 1, '#text': 1 }, 'P in P, no nodes before/after (count)');
+    assert.equal(serializer.serialize(root), '<p class="a"></p><p class="b">123</p><p></p>', 'P in P, splits outer P');
+    assert.deepEqual(countNodes(root), { 'body': 1, 'p': 3, '#text': 1 }, 'P in P, splits outer P (count)');
 
     parser = DomParser({}, schema);
     root = parser.parse('<p class="a">a<p class="b">b</p><p class="c">c</p>d</p>');
     assert.equal(
       serializer.serialize(root),
-      '<p class="a">a</p><p class="b">b</p><p class="c">c</p><p class="a">d</p>',
-      'Two P in P, no nodes before/after'
+      '<p class="a">a</p><p class="b">b</p><p class="c">c</p>d<p></p>',
```

Comment:
Yep, that's correct as that's how the HTML spec says to parse it... basically when it sees a nested `<p>` it closes the current paragraph. Then when it sees `</p>` it says, oh there should have been a paragraph there so it creates an empty one. So since `d` isn't inside a paragraph since the one that was around `a` was closed it ends up being outside.